test/functional/lua/fs_spec.lua

```diff
@@ -416,4 +416,46 @@ describe('vim.fs', function()
       end)
     end)
   end)
+
+  describe('abspath', function()
+    local cwd = vim.uv.cwd()
+    local home = vim.uv.os_homedir()
+
+    assert(cwd ~= nil)
+    assert(home ~= nil)
+
+    if is_os('win') then
+      cwd = cwd:gsub('\\', '/')
+      home = home:gsub('\\', '/')
+    end
```

Comment:
(untested)
```suggestion
    local cwd = assert(is_os('win') and (vim.uv.cwd()):gsub('\\', '/') or vim.uv.cwd())
    local home = assert(is_os('win') and (vim.uv.os_homedir()):gsub('\\', '/') or vim.uv.os_homedir())
```

Reply:
NOTE: cwd and home can be nil, but if we want to not consider that then that's fine

Reply:
If `cwd` or `home` are `nil`, `gsub` will not work

Reply:
> If `cwd` or `home` are `nil`, `gsub` will not work

it will error, just like the assert(). for the purposes of this test code that's pretty equivalent. though I don't know of LuaLS will like it.

Reply:
LuaLS does complain about it if I change it to that, it also looks very cluttered and hard to parse mentally, imo.

Reply:
How is following a bunch of reassignments easier to parse mentally?

Ternaries are preferred. Not going to argue about this over and over.